opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -104,76 +117,116 @@ message Metric {
+  //   repeated RawDataPoint raw_data_points = 3;
 }
 
 // Defines a metric type and its schema.
 message MetricDescriptor {
-  // name of the metric, including its DNS name prefix. It must be unique.
+  // Name of the metric, including its DNS name prefix. It must be unique.
   string name = 1;
 
-  // description of the metric, which can be used in documentation.
+  // Description of the metric, which can be used in documentation.
   string description = 2;
 
-  // unit in which the metric value is reported. Follows the format
+  // Unit in which the metric value is reported. Follows the format
   // described by http://unitsofmeasure.org/ucum.html.
   string unit = 3;
 
-  // Type is the type of values a metric has.
-  enum Type {
-    // INVALID_TYPE is the default Type, it MUST not be used.
-    INVALID_TYPE = 0;
+  // ValueType is the type of values a metric has.  ValueType
+  // determines which field of the DataPoint will be used for Metrics
+  // with this descriptor.  See the definition of Kind for detail on
+  // which aggregations may be expressed for which instruments using
+  // which value types.
+  enum ValueType {
+    // INVALID_TYPE is the default ValueType, it MUST not be used.
+    INVALID_VALUE_TYPE = 0;
 
-    // INT64 values are signed 64-bit integers.
-    //
-    // A Metric of this Type MUST store its values as Int64DataPoint.
-    INT64 = 1;
+    // ScalarInt64 implies that the value is found in
+    // Metric.int64_data_points[].value.
+    SCALAR_INT64 = 1;
 
-    // MONOTONIC_INT64 values are monotonically increasing signed 64-bit
-    // integers.
-    //
-    // A Metric of this Type MUST store its values as Int64DataPoint.
-    MONOTONIC_INT64 = 2;
+    // ScalarDouble implies that the value is found in
+    // Metric.double_data_points[].value.
+    SCALAR_DOUBLE = 2;
 
-    // DOUBLE values are double-precision floating-point numbers.
-    //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    DOUBLE = 3;
+    // Histogram implies that the value is found in
+    // Metric.histogram_data_points[].histogram.
+    HISTOGRAM = 3;
 
-    // MONOTONIC_DOUBLE values are monotonically increasing double-precision
-    // floating-point numbers.
-    //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    MONOTONIC_DOUBLE = 4;
-
-    // Histogram measurement.
-    // Corresponding values are stored in HistogramDataPoint.
-    HISTOGRAM = 5;
-
-    // Summary value. Some frameworks implemented Histograms as a summary of observations
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable percentiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+    // Summary implies that the value is found in
+    // Metric.summary_data_points[].summary.
+    SUMMARY = 4;
```

Comment:
Thinking ahead to the exemplars changes; in #162 we had `SUMMARY_INT64` and `SUMMARY_DOUBLE` which lets us determine the data type of exemplars (int or double). But if we only specify `double` types for summary, does it make sense to have a `SUMMARY_INT64` and `SUMMARY_DOUBLE` type for only exemplars or should there be a different enum for raw measurement type? Same thing for histogram

Reply:
I was on the fence about creating _INT64 and _DOUBLE variations of Histogram and Summary data points. I wouldn't object, but I fear this is more detail than most people want. I draw a distinction between single-value data points (SCALAR, RAW) and multi-value data points (HISTOGRAM, SUMMARY). I think it's more OK to support only floats for multi-value data points, basically, and really not OK to do that for scalars and raw values. 

I asked around whether this was a real problem at Google, and in cases it was. If you count your network traffic in bits per second, your router is likely to overflow a 2^50 counter but not a 2^64 counter.